docs/pages/guides/deep-linking.mdx

```diff
@@ -317,6 +380,38 @@ If you are using [EAS Build](/build/introduction/), you will have to create a ne
 
 </Collapsible>
 
+## Debugging
+
+Expo CLI enables you to test your universal links without deploying a website. Utilizing the [`--tunnel` functionality](/more/expo-cli/#tunneling), you can forward your dev server to a publicly available https URL.
+
+1. Set the environment variable `EXPO_TUNNEL_SUBDOMAIN=my-custom-domain` where "my-custom-domain" is a unique string that you will use during development. This will ensure that your tunnel URL is consistent across dev server restarts.
+2. Setup universal links as described above, but this time using an Ngrok URL: `my-custom-domain.ngrok.io`
+3. Start your dev server with the `--tunnel` flag:
+
+<Terminal cmd={['$ npx expo start --tunnel --dev-client']} />
+
+4. Compile the development build on your device:
+
+<Terminal
+  cmd={[
+    '# Build your native Android project',
+    '$ npx expo run:android',
+    '',
+    '# Build your native iOS project',
+    '$ npx expo run:ios',
+  ]}
+/>
+
+## Trouble Shooting
+
+- Ensure your apple app site association file is valid by using a [validator tool](https://branch.io/resources/aasa-validator/).
+- Ensure your website is served over HTTPS.
+- The uncompressed `apple-app-site-association` file cannot be [larger than 128kb](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html).
+- [Verify Android app links](https://developer.android.com/training/app-links/verify-android-applinks)
+- Ensure both website verification files are served with `content-type` `application/json`.
+- Android verification may take up to 20 seconds to take effect.
+- If you update your web files, be sure to rebuild the native app to trigger a server update on the vendor-side (Apple, Google).
```

Comment:
```suggestion
- If you update your web files, rebuild the native app to trigger a server update on the vendor side (Google, Apple).
```

Suggestion to be concise and referencing [Android, ios](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#referencing-android-ios-and-web) as per writing style guide.